app/shared/environment.js

```diff
@@ -0,0 +1,49 @@
+/*
+ Copyright 2016 Mozilla
+
+ Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ this file except in compliance with the License. You may obtain a copy of the
+ License at http://www.apache.org/licenses/LICENSE-2.0
+ Unless required by applicable law or agreed to in writing, software distributed
+ under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ specific language governing permissions and limitations under the License.
+ */
+/* global LIBDIR */
+
+import yargs from 'yargs';
+import path from 'path';
+import AppDirectory from 'appdirectory';
+import fs from 'fs-extra';
+import manifest from '../../package.json';
+import BUILD_CONFIG from '../../build-config.json';
+
+export const argparser = yargs.usage('Usage: $0 [options]')
```

Comment:
The goal I was going for was that multiple modules could add their arguments to the parser then get the results so I was intentionally having a singleton argparser for the process. I guess now I think of it there is no real reason that is needed, each module could just have its own argparser if it needs one, yargs doesn't modify process.argv I guess.